specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/networkInterfaces.json

```diff
@@ -0,0 +1,519 @@
+          },
+          "description": "IPConfigurations - A list of IPConfigurations of the network interface.",
+          "x-ms-identifiers": []
+        },
+        "macAddress": {
+          "type": "string",
+          "description": "MacAddress - The MAC address of the network interface."
+        },
+        "dnsSettings": {
+          "$ref": "#/definitions/InterfaceDNSSettings",
+          "description": "DNS Settings for the interface"
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Provisioning state of the network interface.",
+          "readOnly": true,
+          "enum": [
+            "Succeeded",
+            "Failed",
+            "InProgress",
+            "Accepted",
+            "Deleting",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningStateEnum",
+            "modelAsString": true
+          }
+        },
+        "status": {
+          "$ref": "#/definitions/NetworkInterfaceStatus"
+        }
+      },
+      "description": "Properties under the network interface resource"
+    },
+    "NetworkInterfaceStatus": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "readOnly": true,
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "description": "NetworkInterface provisioning error code"
+        },
+        "errorMessage": {
+          "type": "string",
+          "description": "Descriptive error message"
+        },
+        "provisioningStatus": {
+          "type": "object",
+          "properties": {
+            "operationId": {
+              "type": "string",
+              "description": "The ID of the operation performed on the network interface"
+            },
+            "status": {
+              "type": "string",
+              "description": "The status of the operation performed on the network interface [Succeeded, Failed, InProgress]",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "InProgress"
+              ],
+              "x-ms-enum": {
+                "name": "Status",
+                "modelAsString": true
+              }
+            }
+          }
+        }
+      },
+      "description": "The observed state of network interfaces"
+    },
+    "IPConfiguration": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "name": {
+          "type": "string",
+          "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-._a-zA-Z0-9]{0,78}[_a-zA-Z0-9]$",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ],
+          "description": "Name - The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "properties": {
+          "type": "object",
+          "properties": {
+            "gateway": {
+              "x-ms-client-flatten": true,
```

Comment:
>     "x-ms-client-flatten": true, [](http://example.com/codeflow?start=9&length=33)

nit - is this x-ms-client-flatten even meaningful on the property level when its not type: object? I don't think it is?

Reply:
will fix

Reply:
yep makes sense to remove. Will fix